pkg/exec_unit/plugin_assets.go

```diff
@@ -75,6 +84,45 @@ type assetPathMatcher struct {
 	err     error
 }
 
+func (m *assetPathMatcher) ModifyPathsForAnnotatedFile(path string) error {
```

Comment:
nit: Having this be a mutator function seems a bit weird to me. An assetPathMatcher isn't valid before calling this function and I'm not sure if it'd behave right if called twice, so callers need to make sure to call it exactly once on construction. Seems like it might be better served in a `NewAssetPathMatcher` constructor function.